src/analytics/AnalyticTable.js

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { Col, Row, InputGroup, InputGroupAddon, InputGroupText, CustomInput } from 'reactstrap';
+
+import AsfanDropdown from 'components/AsfanDropdown';
+import ErrorBoundary from 'components/ErrorBoundary';
+import { SortableTable, compareStrings, valueRenderer } from 'components/SortableTable';
+import useQueryParam from 'hooks/useQueryParam';
+import CardPropType from 'proptypes/CardPropType';
+import CubePropType from 'proptypes/CubePropType';
+import { sortIntoGroups, SORTS, getLabels, cardCanBeSorted } from 'utils/Sort';
+import { fromEntries } from 'utils/Util';
+
+const sortWithTotal = (pool, sort) => {
+  const groups = sortIntoGroups(pool, sort);
+
+  for (const key of Object.keys(groups)) {
+    groups[key] = groups[key].reduce((acc, card) => acc + card.asfan, 0);
+  }
+
+  groups.Total = pool.reduce((acc, card) => acc + card.asfan, 0);
+
+  return groups;
+};
+
+const AnalyticTable = ({ cards: allCards, cube, defaultFormatId, setAsfans }) => {
+  const [primary, setPrimary] = useQueryParam('primary', 'Color Identity');
+  const [secondary, setSecondary] = useQueryParam('secondary', 'Type');
+
+  // some criteria cannot be applied to some cards
+  const cards = useMemo(
+    () => allCards.filter((card) => cardCanBeSorted(card, primary) && cardCanBeSorted(card, secondary)),
+    [allCards, primary, secondary],
+  );
+  const primaryGroups = useMemo(() => {
+    const groups = sortIntoGroups(cards, primary);
+    groups.Total = {};
+    for (const key of Object.keys(groups)) {
+      if (key !== 'Total') {
+        groups[key] = sortWithTotal(groups[key], secondary);
+        for (const subkey of Object.keys(groups[key])) {
+          if (!groups.Total[subkey]) {
+            groups.Total[subkey] = 0;
+          }
+          groups.Total[subkey] += groups[key][subkey];
+        }
+      }
+    }
+    return groups;
+  }, [cards, primary, secondary]);
+  const entryRenderer = useCallback(
+    (value) => (
+      <>
+        {valueRenderer(value ?? 0)}
+        <span className="percent">{valueRenderer(((value ?? 0) / primaryGroups.Total.Total) * 100)}%</span>
```

Comment:
They do basically mean the same thing though if you think about it a certain way. Reading it right to left you can think of it saying `sumOfOtherCells(pointwiseSumOfOtherRows(primaryGroups)` or alternatively as `pointwiseSum(pointwiseSum(primaryGroups))` where each call reduces the dimension it's provided with by 1. Alternatively it reads as `Total column of the Total row from primaryGroups` which is the main reason I'd like to leave it since it'll sync up with how it is visually displayed.

Reply:
Sure. I don't mind it all that much, it just stood out to me.